gnomad_qc/v3/variant_qc/random_forest.py

```diff
@@ -0,0 +1,475 @@
+    )
+    actions.add_argument(
+        "--annotate_for_rf",
+        help="Creates an annotated ht with features for RF",
+        action="store_true",
+    )
+    actions.add_argument("--train_rf", help="Trains RF model", action="store_true")
+    actions.add_argument(
+        "--apply_rf", help="Applies RF model to the data", action="store_true"
+    )
+    actions.add_argument("--finalize", help="Write final RF model", action="store_true")
+
+    annotate_params = parser.add_argument_group("Annotate features parameters")
+    annotate_params.add_argument(
+        "--impute_features_no_variant_type",
+        help="If set, feature imputation is NOT split by variant type.",
+        action="store_true",
+    )
+    annotate_params.add_argument(
+        "--n_partitions",
+        help="Desired number of partitions for annotated RF Table",
+        type=int,
+        default=5000,
+    )
+
+    rf_params = parser.add_argument_group("Random Forest parameters")
+    rf_params.add_argument(
+        "--fp_to_tp",
+        help="Ratio of FPs to TPs for training the RF model. If 0, all training examples are used. (default=1.0)",
+        default=1.0,
+        type=float,
+    )
+    rf_params.add_argument(
+        "--test_intervals",
+        help='The specified interval(s) will be held out for testing and evaluation only. (default to "chr20")',
+        nargs="+",
+        type=str,
+        default="chr20",
+    )
+    rf_params.add_argument(
+        "--num_trees",
+        help="Number of trees in the RF model. (default=500)",
+        default=500,
+        type=int,
+    )
+    rf_params.add_argument(
+        "--max_depth",
+        help="Maxmimum tree depth in the RF model. (default=5)",
+        default=5,
+        type=int,
+    )
+
+    training_params = parser.add_argument_group("Training data parameters")
+    training_params.add_argument(
+        "--adj", help="Use adj genotypes.", action="store_true"
+    )
+    training_params.add_argument(
+        "--vqsr_training", help="Use VQSR training examples", action="store_true"
+    )
+    training_params.add_argument(
+        "--vqsr_type",
+        help="If a string is provided the VQSR training annotations will be used for training.",
+        default="alleleSpecificTrans",
+        choices=["classic", "alleleSpecific", "alleleSpecificTrans"],
+        type=str,
+    )
+    training_params.add_argument(
+        "--no_transmitted_singletons",
+        help="Do not use transmitted singletons for training.",
+        action="store_true",
+    )
+    training_params.add_argument(
+        "--no_inbreeding_coeff",
+        help="Train RF without inbreeding coefficient as a feature.",
+        action="store_true",
+    )
+    training_params.add_argument(
+        "--filter_centromere_telomere",
+        help="Train RF without centromeres and telomeres.",
+        action="store_true",
+    )
+
+    finalize_params = parser.add_argument_group("Finalize RF Table parameters")
```

Comment:
Remove all finalization parameters; this activity has been moved to a separate script